docs/src/main/sphinx/functions/table.rst

```diff
@@ -0,0 +1,69 @@
+===============
+Table functions
+===============
+
+A table function is a function returning a table. It can be invoked inside the
+``FROM`` clause of a query::
+
+    SELECT * FROM my_function(1, 100)
+
+The row type of the returned table can depend on the arguments passed with
+invocation of the function. If different row types can be returned, the
+function is a **polymorphic table function**.
+
+Polymorphic table functions allow you to dynamically invoke custom logic from
+within the SQL query. They can be used for working with external systems as
+well as for enhancing Trino with capabilities going beyond the SQL standard.
+
+Trino supports adding custom table functions. They are declared by connectors
```

Comment:
we need to add info about what table functions are there for each connector .. even if that is none 

and then be explicit about linking to the dev guide to figure out how to add any